core/sched.c

```diff
@@ -220,6 +224,19 @@ void sched_set_status(thread_t *process, thread_status_t status)
             clist_rpush(&sched_runqueues[process->priority],
                         &(process->rq_entry));
             _set_runqueue_bit(process);
+
+            /* some thread entered a runqueue
+             * if it is the active runqueue
+             * inform the runqueue_change callback */
+            /* might be more efficient to check using
```

Comment:
that if in the comment is a bit simplified and may not handle edgecases, but maybe we can do away with the if (since the callback has all the checks), The if is there to keep us from jumping through our program-memory (avoiding stress to caches and prefetch).

Reply:
then I don't understand what the comment ("might be more efficient ...") means / suggests.

Reply:
the matching if would be `if(_get_prio_queue_from_runqueue() == process->priority)` (that one does not ensure there is something active right now contrary to the statement used right now)

Reply:
while i think i know how much work `thread_get_active()` does (nearly nothing, but i need some member of it and check if i got a result) i don't know if `_get_prio_queue_from_runqueue()` is faster (maybe much more arch dependent)

Reply:
` if(_get_prio_queue_from_runqueue() == process->priority) { ` yielded bigger binary size for native and some stm32-L0 ->  away with that comment